cogboard-webapp/src/components/widgets/types/LogViewerWidget/LogList/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { useRef, useEffect, useState } from 'react';
+import {
+  getGridTemplate,
+  filterByRegExp,
+  filterByDateSpan,
+  isLogHighlighted,
+  filterByLevel
+} from './helpers';
+import { getFilters, getLevel } from '../Toolbar/FilterPicker/helpers';
+import { getDateSpan } from '../Toolbar/DateRangePicker/helpers';
+import { useTheme } from '@material-ui/core';
+import LogEntry from './LogEntry';
+import {
+  Container,
+  Header,
+  GridSchema,
+  ColumnTitle,
+  LogsWrapper,
+  StyledVirtuoso,
+  VariableGridSchema
+} from './styled';
+
+export default function LogList({
+  widgetLocalStorage,
+  logs,
+  template,
+  search,
+  shouldFollowLogs,
+  handleFollowChange
+}) {
+  const theme = useTheme();
+  const scrollerRef = useRef(null);
+  const [scroll, setScroll] = useState(0);
+
+  const filters = getFilters(widgetLocalStorage);
+  const level = getLevel(widgetLocalStorage);
+  const dateSpan = getDateSpan(widgetLocalStorage);
+
+  const filteredLogs = logs
+    ?.filter(log => filterByRegExp(log, filters))
+    .filter(log => filterByDateSpan(log, dateSpan))
+    .filter(log => filterByLevel(log, level))
```

Comment:
Fix: Change the order of how filters are applied. In my opinion, the regexp check has the most performance impact.

1. Check Level
2. Check dates
3. Check regexp